### PR TITLE
refactor: split ocr_extract_text from ebook

### PR DESCRIPTION
**Pull request by yatli**

### CODE REVIEW

File: third-party/libscan/libscan/ebook/ebook.c [hunk 1]
```diff
@@ -225,34 +223,19 @@ static int read_stext_block(fz_stext_block *block, text_buffer_t *tex) {
...
                 UNUSED(fz_color_params color_params)) {
 
     int l2factor = 0;
 
-    if (img->w > MIN_OCR_SIZE && img->h > MIN_OCR_SIZE && IS_VALID_BPP(img->n)) {
-
+    if (img->w > MIN_OCR_SIZE && img->h > MIN_OCR_SIZE && OCR_IS_VALID_BPP(img->n)) {
         fz_pixmap *pix = img->get_pixmap(fzctx, img, NULL, img->w, img->h, &l2factor);
-
-        if (pix->h > MIN_OCR_SIZE && img->h > MIN_OCR_SIZE && img->xres != 0) {
```
Comment (yatli) [inline third-party/libscan/libscan/ebook/ebook.c:240]: typo here? img->w, img->h in outer loop, so pix->w, pix->h here?

File: third-party/libscan/libscan/ocr/ocr.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#define MIN_OCR_SIZE 350
+#define MIN_OCR_LEN 10
+
+#define OCR_IS_VALID_BPP(d)                                                    \
+  ((d) == 1 || (d) == 2 || (d) == 4 || (d) == 8 || (d) == 16 || (d) == 24 ||   \
+   (d) == 32)
+
+typedef void (*ocr_extract_callback_t)(const char *, size_t);
+
+__always_inline static void
```
[the remark this answers is not in the document]
Comment (simon987) [reply third-party/libscan/libscan/ocr/ocr.h:16]: I will need to test this but I think that one can specify multiple languages for tesseract with a `+` separator `eng+chi-sim`

[the remark this answers is not in the document]
Comment (yatli) [reply third-party/libscan/libscan/ocr/ocr.h:16]: lol yeah I just tested it. The caveat is to `sudo touch /usr/share/tessdata/eng+chi_sim.traineddata`

[the remark this answers is not in the document]
Comment (yatli) [reply third-party/libscan/libscan/ocr/ocr.h:16]: There's a "Could not find tesseract language file" somewhere...

[the remark this answers is not in the document]
Comment (simon987) [reply third-party/libscan/libscan/ocr/ocr.h:16]: For #217 I think you have generalized the extracted function such that I can just decode a picture with Ffmpeg into an RGB buffer and run it

IIRC there's a check during the CLI args validation to see if the language file exists, I'll have to modify it to handle multiple languages

File: third-party/libscan/libscan/ebook/ebook.c (hunk 1 above)
Comment (simon987) [reply third-party/libscan/libscan/ebook/ebook.c:240]: Good catch, you're right it should be `if (pix->w > MIN_OCR_SIZE && pix->h > MIN_OCR_SIZE && img->xres != 0) {`

**Comment by simon987** [comment]
Thanks a lot for the help! :heart:  I appreciate it

**Pull request merged**